engine/api/src/org/apache/cloudstack/engine/cloud/entity/api/VirtualMachineEntity.java

```diff
@@ -115,6 +115,12 @@ String reserve(DeploymentPlanner plannerToUse, @BeanParam DeploymentPlan plan, E
     boolean stop(String caller) throws ResourceUnavailableException, CloudException;
 
     /**
+     * Stop the virtual machine, optionally force
+     *
+     */
+    boolean stop(String caller, boolean forced) throws ResourceUnavailableException, CloudException;
```

Comment:
@nathanejohnson I wasn't thinking to change any of the existing code.  I am just interested in avoiding incurring any more flag argument technical debt.

In terms of testing, this change is not hypervisor specific.  Therefore, a test case that utilizes the simulator might be a more fruitful path.  @rhtyd may have some thoughts as to how you could accomplish it.